plugin/pkg/admission/serviceplan/defaultserviceplan/admission.go

```diff
@@ -196,6 +283,30 @@ func (d *defaultServicePlan) getClusterServiceClassByField(a admission.Attribute
 	return nil, admission.NewNotFound(a)
 }
 
+func (d *defaultServicePlan) getServiceClassByField(a admission.Attributes, ref *servicecatalog.PlanReference) (*servicecatalog.ServiceClass, error) {
+	filterField := ref.GetClassFilterFieldName()
```

Comment:
GetClassFilterFieldName() - this is currently returning a ClusterServiceClass - I think you need one that is ns scoped.  There are other calls to this that look like they should be NS scoped as well.

I could be wrong about this.  If I'm right, I think there may be a similar issue with GetPlanFilterFieldName() - its cluster specific and you need a NS scoped one?

Reply:
Yep, think you're totally right. Nice catch, will update.